File: configlibrary/src/main/java/com/shmedo/configlibrary/ble/cmd/entity/ServerNumberEntity.java
package com.shmedo.configlibrary.ble.cmd.entity;

import com.shmedo.configlibrary.ble.interfaces.Validater;

/**
 * Created by adu on 2017/12/14.
 * 获取服务器(数据中心)编号参数
 */
public class ServerNumberEntity implements Validater {
    private int number;

    public ServerNumberEntity(int number) {
        this.number = number;
    }

    @Override
    public void validate() {
    }

    @Override
    public String toString() {
        return String.valueOf(this.number);
    }
}

File: configlibrary/src/main/java/com/shmedo/configlibrary/iot/cmd/entity/ServerNumberEntity.kt
package com.shmedo.configlibrary.iot.cmd.entity

import com.shmedo.configlibrary.ble.exception.DASParameterException
import com.shmedo.configlibrary.ble.interfaces.Validater

/**
 * 创建者:   gonghe <br></br>
 * 创建时间:  11/16/20 <br></br>
 * 描述：     获取服务器(数据中心)编号参数
 */
class ServerNumberEntity(private val number: Int) : Validater {
    override fun validate() {
        if (number != 1 && number != 2 && number != 3 && number != 4) throw DASParameterException("服务器编号不存在")
    }

    override fun toString(): String {
        return "centerid=$number"
    }
}